common/src/main/kotlin/org/valkyrienskies/mod/common/util/RelocationUtil.kt

```diff
@@ -0,0 +1,21 @@
+package org.valkyrienskies.mod.common.util
+
+import net.minecraft.core.BlockPos
+import net.minecraft.world.level.Level
+import net.minecraft.world.level.chunk.LevelChunk
+
+// the from and the to can be local or global
+fun relocateBlock(fromChunk: LevelChunk, from: BlockPos, toChunk: LevelChunk, to: BlockPos) {
+    val state = fromChunk.getBlockState(from)
+    val entity = fromChunk.getBlockEntity(from)
+
+    toChunk.setBlockState(to, state, false) // TODO should isMoving be false?
+    entity?.let {
```

Comment:
I think the safest way to do this is to save the entity to a CompoundTag, then manually edit the position thats saved, and then load the new BlockEntity from the modified tag.

We can also add a relocation API to allow mods to fix bugs with relocated blocks.